src/log.c

```diff
@@ -592,24 +601,22 @@ qd_error_t qd_log_entity(qd_entity_t *entity)
             QD_ERROR_BREAK();
         }
 
-        //
-        // Obtain the log_source_lock lock. This lock is also used when write_log() is called.
-        //
-        sys_mutex_lock(log_source_lock);
-
-        qd_log_source_t *src = qd_log_source_lh(module); /* The original(already existing) log source */
+        qd_log_source_t *src = qd_log_source_find_or_make(module, false); /* The original (already existing) log source */
 
         if (has_output_file) {
+            sys_mutex_lock(log_source_lock);
             log_sink_t* sink = log_sink_lh(outputFile);
+            sys_mutex_unlock(log_source_lock);
```

Comment:
What prevents another thread from calling this function and freeing the same sink (line 618) once the first thread drops the lock?